src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -631,4 +643,130 @@ public static void updateConfigErrorMessage(Config result, String key, String ms
+            case BOOLEAN:
+              schemaMap.put(field.name(), "boolean");
+              break;
+            case BYTES:
+              schemaMap.put(field.name(), "binary");
+              break;
+            case DOUBLE:
+              schemaMap.put(field.name(), "double");
+              break;
+            case FLOAT:
+              schemaMap.put(field.name(), "float");
+              break;
+            case INT:
+              schemaMap.put(field.name(), "int");
+              break;
+            case LONG:
+              schemaMap.put(field.name(), "number");
+              break;
+            case STRING:
+              schemaMap.put(field.name(), "string");
+              break;
+            case ARRAY:
+              schemaMap.put(field.name(), "array");
+              break;
+            default:
+              schemaMap.put(field.name(), "variant");
+          }
+        }
+      }
+    }
+    return schemaMap;
+  }
+
+  /**
+   * From the connector config extract whether the avro value converter is used
+   *
+   * @param connectorConfig
+   * @return whether the avro value converter is used
+   */
+  public static boolean usesAvroValueConverter(final Map<String, String> connectorConfig) {
+    List<String> validAvroConverter = new ArrayList<>();
+    validAvroConverter.add("io.confluent.connect.avro.AvroConverter");
+    validAvroConverter.add("com.snowflake.kafka.connector.records.SnowflakeAvroConverter");
+    validAvroConverter.add("io.confluent.encryption.connect.SecuredAvroConverter");
+    // the last one in doubt
+    if (connectorConfig.containsKey(SnowflakeSinkConnectorConfig.VALUE_CONVERTER_CONFIG_FIELD)) {
+      String valueConverter =
+          connectorConfig.get(SnowflakeSinkConnectorConfig.VALUE_CONVERTER_CONFIG_FIELD);
+      return validAvroConverter.contains(valueConverter);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * From get the schemaMap for the table from topics. Topics will be collected from topicToTableMap
+   * or connectorConfig
+   *
+   * @param topicToTableMap
+   * @param connectorConfig
+   * @return the map from the columnName to their type
+   */
+  public static Map<String, String> getSchemaMap(
+      final Map<String, String> topicToTableMap, final Map<String, String> connectorConfig) {
+    Map<String, String> schemaMap = new HashMap<>();
+    if (!topicToTableMap.isEmpty()) {
+      for (String topic : topicToTableMap.keySet()) {
+        Map<String, String> tempMap =
+            Utils.getValueSchemaFromSchemaRegistryURL(
+                topic, connectorConfig.get("value.converter.schema.registry.url"));
+        schemaMap.putAll(tempMap);
+      }
+    } else {
+      // if topic is not present in topic2table map, the table name must be the same with the
+      // topic
+      schemaMap =
+          Utils.getValueSchemaFromSchemaRegistryURL(
+              connectorConfig.get("topics"),
+              connectorConfig.get("value.converter.schema.registry.url"));
```

Comment:
can you extract `value.converter.schema.registry.url` into a public static final string in constants.java file
I see this being used at multiple places.

Reply:
Done